src/main/java/uk/gov/pay/api/exception/mapper/ViolationExceptionMapper.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.pay.api.exception.mapper;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.jersey.validation.JerseyViolationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.api.model.PaymentError;
+
+import javax.annotation.Priority;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static uk.gov.pay.api.model.PaymentError.Code.CREATE_PAYMENT_VALIDATION_ERROR;
+
+@Priority(1)
+public class ViolationExceptionMapper implements ExceptionMapper<JerseyViolationException> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViolationExceptionMapper.class);
+
+    @Override
+    public Response toResponse(JerseyViolationException exception) {
+        LOGGER.error(exception.getMessage());
+        ConstraintViolation<?> firstException = getFirstException(exception);
+        String message = firstException.getMessage();
+        String fieldName = getApiFieldName(firstException);
+        PaymentError paymentError = PaymentError.aPaymentError(fieldName, CREATE_PAYMENT_VALIDATION_ERROR, message);
+
+        return Response.status(422)
+                .entity(paymentError)
+                .build();
+    }
+
+    private String getApiFieldName(ConstraintViolation<?> firstException) {
+        Field field = getField(firstException);
+        return getJacksonPropertyName(field).orElse(field.getName());
+    }
+
+    private Optional<String> getJacksonPropertyName(Field field) {
+        return Arrays.stream(field.getAnnotations())
+                .filter(annotation -> annotation instanceof JsonProperty)
+                .findFirst()
+                .map(annotation -> ((JsonProperty) annotation).value());
+    }
+    
+    private Field getField(ConstraintViolation<?> firstException) {
+        Class<?> leafBean = firstException.getLeafBean().getClass();
+        String fieldName = getLastElementOfPath(firstException.getPropertyPath());
```

Comment:
Purely down to my unfamiliarity with the javax.validation API. Why is the last element of the path got here and not the first?

Reply:
The last element is the field name, the path will look something like `CreatePaymentRequest.amount` or `CreatePaymentRequest.returnUrl`. Since it isn't clear, perhaps I'll rename this `getFieldNameFromPath`?

Reply:
`getFieldNameFromPath` would be clearer. You'd still be doing `pathParts[pathParts.length - 1]` to get this though; I wouldn't be averse to writing a short comment to make it clear why we're doing this. Unfortunately this is a side effect of using reflection....anything goes!